streams/src/main/java/org/apache/kafka/streams/state/Stores.java

```diff
@@ -44,6 +44,40 @@
 
 /**
  * Factory for creating state stores in Kafka Streams.
+ * <p>
+ * For high-level DSL, i.e., {@link org.apache.kafka.streams.StreamsBuilder StreamsBuilder}, users create
+ * {@link StoreSupplier}s that can be further customized via
+ * {@link org.apache.kafka.streams.kstream.Materialized Materialized}.
+ * For example, a topic read as {@link org.apache.kafka.streams.kstream.KTable KTable} can be materialized into an
+ * in-memory store with custom key/value serdes and caching disabled:
+ * <pre>{@code
+ * StreamsBuilder builder = new StreamsBuilder();
+ * KeyValueBytesStoreSupplier storeSupplier = Stores.inMemoryKeyValueStore("queryable-store-name");
+ * KTable<Long,String> table = builder.table(
+ *   "topicName",
+ *   Materialized.as(storeSupplier)
+ *               .withKeySerde(Serdes.Long())
+ *               .withValueSerde(Serdes.String())
+ *               .withCachingDisabled());
+ * }</pre>
+ * For low-level Processor API, i.e., {@link org.apache.kafka.streams.Topology Topology}, users create
+ * {@link StoreBuilder}s that can be attached to {@link org.apache.kafka.streams.processor.Processor Processor}s.
+ * For example, you can create an {@link org.apache.kafka.streams.kstream.Windowed windowed} RocksDB store with custom
+ * changelog configuration like:
+ * <pre>{@code
+ * Topology topology = new Topology();
+ * topology.addProcessor("processorName", ...);
+ *
+ * Map<String,String> changelogConfig = new HashMap<>();
```

Comment:
Suggestion:  I am not sure which terminology we use elsewhere, but "changelogConfig" may suggest to the average reader that there's a separate "changelog" concept that they may have never heard of.  It's simply a topic configuration.

I'd suggest to rename `changelogConfig` to `changelogTopicConfig` or the shorter `topicConfig`.